core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -123,6 +129,39 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function. */
+  public static String toBase64(String string) {
+    return toBase64_(string.getBytes(UTF_8));
+  }
+
+  /** SQL TO_BASE64(string) function for binary string. */
+  public static String toBase64(ByteString string) {
+    return toBase64_(string.getBytes());
+  }
+
+  private static String toBase64_(byte[] bytes) {
+    String base64 = Base64.getEncoder().encodeToString(bytes);
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    if (base64.length() > 76) {
+      Splitter.fixedLength(76).split(base64).iterator().forEachRemaining(s -> {
+        str.append(s);
+        str.append("\n");
+      });
+      return str.substring(0, str.length() - 1);
+    }
+    return base64;
+  }
+
+  /** SQL FROM_BASE64(string) function. */
+  public static ByteString fromBase64(String base64) {
+    try {
+      base64 = FROM_BASE64_REGEXP.matcher(base64).replaceAll("");
```

Comment:
Just wondering: could you please explain why `java.util.Base64#getMimeDecoder` does not work here?
It looks like `java.util.Base64#getMimeDecoder` is just enough, and it does not require regexp preprocessing

Reply:
When I using Base64.getMimeDecoder().decode(base64), the result of from_base64('-100') is d74d, however it should be null in MySql.